From 61b52e46ecea59f047c68e553af33fdd1e54e642 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 6 Dec 2017 21:32:55 +0900
Subject: [PATCH 304/909] drm/bridge: dumb-vga-dac: Add preferred resolution
 option

commit d22143d3cce712d1bae7bc88dff67e4f68ad7277 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/dumb-vga-dac.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/dumb-vga-dac.c b/drivers/gpu/drm/bridge/dumb-vga-dac.c
index 0894939a8966..8590e73590b2 100644
--- a/drivers/gpu/drm/bridge/dumb-vga-dac.c
+++ b/drivers/gpu/drm/bridge/dumb-vga-dac.c
@@ -28,6 +28,8 @@ struct dumb_vga {
 	struct i2c_adapter	*ddc;
 	struct regulator	*vdd;
 	bool			ddc_flag;
+	u32			width;
+	u32			height;
 };
 
 static inline struct dumb_vga *
@@ -70,7 +72,7 @@ static int dumb_vga_get_modes(struct drm_connector *connector)
 	ret = drm_add_modes_noedid(connector, 1920, 1200);
 
 	/* And prefer a mode pretty much anyone can handle */
-	drm_set_preferred_mode(connector, 1024, 768);
+	drm_set_preferred_mode(connector, vga->width, vga->height);
 
 	return ret;
 }
@@ -178,6 +180,13 @@ static struct i2c_adapter *dumb_vga_retrieve_ddc(struct device *dev)
 	else
 		vga->ddc_flag = true;
 
+	of_property_read_u32(remote, "width", &vga->width);
+	of_property_read_u32(remote, "height", &vga->height);
+	if (vga->width == 0 || vga->height == 0) {
+		vga->width = 1024;
+		vga->height = 768;
+	}
+
 	phandle = of_parse_phandle(remote, "ddc-i2c-bus", 0);
 	of_node_put(remote);
 	if (!phandle)
-- 
2.17.1

